### PR TITLE
Add Japanese Translation

### PR DESCRIPTION
**Pull request by OctSquid**
I have added a new Japanese translation.

### CODE REVIEW

File: HedgeModManager/Languages/ja-JP.xaml
Comment (RagdollClash) [inline HedgeModManager/Languages/ja-JP.xaml]: Tokyo 2020 has a Japanese name, referring to it's [SteamDB entry](https://steamdb.info/app/981890/info/).
```suggestion
    <system:String x:Key="GameTokyo2020">Olympic Games Tokyo 2020 - 東京2020オリンピック</system:String>
```

**Review by RagdollClash — CHANGES_REQUESTED** [review]

**Review by RagdollClash — COMMENTED** [review]
The file looks fine, but I don't know any Japanese so I'm gonna have to wait for someone else who knows Japanese (or is fluent enough with it) to review it.

File: HedgeModManager/Languages/ja-JP.xaml
Comment (thesupersonic16) [inline HedgeModManager/Languages/ja-JP.xaml]: The game titles are fairly long, are the English titles well known to the Japanese audience? If not, maybe you could change it to something like this. Go with what you think is best.
```suggestion
    <system:String x:Key="GameSonicGenerations">ソニック ジェネレーションズ</system:String>
    <system:String x:Key="GameSonicLostWorld">ソニックロストワールド</system:String>
    <system:String x:Key="GameSonicForces">ソニックフォース</system:String>
    <system:String x:Key="GamePuyoPuyoTetris2">ぷよぷよテトリス ２</system:String>
    <system:String x:Key="GameTokyo2020">東京2020オリンピック</system:String>
    <system:String x:Key="GameSonicColorsUltimate">ソニックカラーズ アルティメット</system:String>
    <system:String x:Key="GameSonicOrigins">ソニックオリジンズ</system:String>
    <system:String x:Key="GameSonicFrontiers">ソニックフロンティア</system:String>
```

File: HedgeModManager/Languages/ja-JP.xaml
Comment (thesupersonic16) [inline HedgeModManager/Languages/ja-JP.xaml]: Minor typo, GitHub is a company name, so the H should to be uppercase.
```suggestion
    <system:String x:Key="ExceptionWindowButtonReport"    >GitHubでレポートを提出する</system:String>
```

File: HedgeModManager/Languages/ja-JP.xaml
Comment (thesupersonic16) [inline HedgeModManager/Languages/ja-JP.xaml]: Is the space here intentional? This results in text like
`6 個のMODをロードしました` 
`ソニックフロンティア を起動中`
`HE2ModLoader をインストールしました`

If this is right, just disregard this review comment and set it as resolve.

File: HedgeModManager/Languages/ja-JP.xaml
Comment (thesupersonic16) [inline HedgeModManager/Languages/ja-JP.xaml:195]: These are not translated. While it is formatted to work in English, can you translate this so that it looks correct in the UI? Maybe in the future we can get this changed to a format string.

**Review by thesupersonic16 — CHANGES_REQUESTED** [review]
Please confirm all the review comments and apply any changes if needed.

There appears to be some kind of inconsistentcy in the use of spaces, can you please ensure it is used correctly?

File: HedgeModManager/Languages/ja-JP.xaml
Comment (OctSquid) [reply HedgeModManager/Languages/ja-JP.xaml]: In Japan, Sonic game titles have Japanese names as a subtitle to the English name of the title.
However, as you say, titles such as `Sonic Colors - ソニックカラーズ` are too long and do not display properly, so I'm going to unify them into Japanese names.

File: HedgeModManager/Languages/ja-JP.xaml
Comment (OctSquid) [reply HedgeModManager/Languages/ja-JP.xaml]: Corrected spaces appropriately.

File: HedgeModManager/Languages/ja-JP.xaml
Comment (OctSquid) [reply HedgeModManager/Languages/ja-JP.xaml:195]: It is difficult to properly translate a sentence like `A made by B on C` without a format string because Japanese grammar differs from English.  
However, words like `made by` and `made on` are widely known in Japan, and it would not be a problem if they were not translated.

Comment (thesupersonic16) [reply HedgeModManager/Languages/ja-JP.xaml:195]: Ok, then we can leave it as is for now.

File: HedgeModManager/Languages/ja-JP.xaml
Comment (thesupersonic16) [reply HedgeModManager/Languages/ja-JP.xaml]: Are L110 and L180 good too? After that I think this is good to be merged.

**Review by thesupersonic16 — APPROVED** [review]

File: HedgeModManager/Languages/ja-JP.xaml
Comment (OctSquid) [reply HedgeModManager/Languages/ja-JP.xaml]: Oh, I had overlooked that.
Both have been corrected.
Thank you.

**Comment by thesupersonic16** [comment]
What name do you want to go by for the translator credits? OctSquid?

**Comment by OctSquid** [comment]
That's ok.

**Comment by thesupersonic16** [comment]
Cool, justing waiting on @ActualMandM's approval now.

**Comment by OctSquid** [comment]
Sorry. I made misclicks.

**Review by RagdollClash — APPROVED** [review]
LGTM

**Pull request merged**
